src/librustc_middle/mir/mod.rs

```diff
@@ -72,15 +72,21 @@ impl<'tcx> HasLocalDecls<'tcx> for Body<'tcx> {
 
 /// The various "big phases" that MIR goes through.
 ///
+/// These phases all describe dialects of MIR. Since all MIR uses the same datastructures, the
+/// dialects forbid certain variants or values in certain phases.
+///
 /// Warning: ordering of variants is significant.
 #[derive(Copy, Clone, RustcEncodable, RustcDecodable, Debug, PartialEq, Eq, PartialOrd, Ord)]
 #[derive(HashStable)]
 pub enum MirPhase {
     Build = 0,
     Const = 1,
     Validated = 2,
+    /// After this phase, the only `AggregateKind`s allowed are `Array` and `Generator`
     DropElab = 3,
-    Optimized = 4,
+    /// After this phase `AggregateKind::Generator` is gone for good.
+    OptimizedWithGenerators = 4,
```

Comment:
Maybe `GeneratorElab` or `GeneratorLowering` or so would be a better name?

Reply:
I renamed it to `GeneratorLowering` and renamed `DropElab` to `DropLowering`